dataactvalidator/validation_handlers/validator.py

```diff
@@ -375,12 +374,12 @@ def rule_greater(cls, data, value, rule, datatype, interfaces, record):
     @classmethod
     def rule_equal(cls, data, value, rule, datatype, interfaces, record):
         """Checks that data is equal to specified value"""
-        return Validator.getType(data,datatype) == Validator.getType(value,datatype)
+        return Validator.getType(data.lower(),datatype) == Validator.getType(value.lower(),datatype)
```

Comment:
Multi-field rules get passed the entire record as data, which is a dict.  So it does not always start as a string, and you'll get an error trying to .lower() a dict.  At a minimum if you don't want to do any extra checking, you could explicitly cast as a str with (str(data).lower()) to avoid those errors.